tokio-console/src/state/mod.rs

```diff
@@ -264,12 +300,50 @@ impl Metadata {
                 .into_iter()
                 .map(|n| strings.string(n))
                 .collect(),
+            name: strings.string(pb.name),
             target: strings.string(pb.target),
+            location: pb
+                .location
+                .map(|pb| Location::from_proto(pb, strings))
+                .unwrap_or_default(),
             id,
         }
     }
 }
 
+// === impl Location ===
+impl Location {
+    fn from_proto(pb: proto::Location, strings: &mut intern::Strings) -> Self {
+        Self {
+            file: pb.file.map(|s| strings.string(s)),
+            module_path: pb.module_path.map(|s| strings.string(s)),
+            line: pb.line,
+            column: pb.column,
+        }
+    }
+}
+
+impl Display for Location {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        if let Some(file) = &self.file {
+            write!(f, "{file}")?;
```

Comment:
the use of this syntax increases the `tokio-console` crate's minimum Rust dependency, so we'll either have to update the MSRV in Cargo.toml and on CI, or use the old syntax.